### ISSUE TITLE
Wish to add that the background of video markers is transparent similar to the green-screen configuration

### ISSUE THREAD

**Issue by tjw-javen** [mistic100/Photo-Sphere-Viewer#1139]
### Describe the feature

like this code   new ShaderMaterial({
                        uniforms: {
                          videoTexture: {
                            value: videoTexture,
                          },
                          threshold: {
                            value: 0.05,
                          },
                        },
                        vertexShader: `
                            varying vec2 vUv;
                            void main() {
                                vUv = uv;
                                gl_Position = projectionMatrix * modelViewMatrix * vec4( position, 1.0 );
                            }`,
                        fragmentShader: `
                        uniform sampler2D videoTexture;
                        uniform float threshold;
                        varying vec2 vUv;
                        void main() {
                            vec4 color = texture2D(videoTexture, vUv);
                            float greenScreen = color.g - max(color.r, color.b);
                            float alpha = 1.0 - smoothstep(threshold - 0.05, threshold + 0.05, greenScreen);
                            gl_FragColor = vec4(color.rgb, alpha);
                        }
                        `,
                        transparent: true,
                        side: DoubleSide,
                      })

**Comment by mistic100** [comment]
Please edit your message with markdown code blocks https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

**Comment by tjw-javen** [comment]
```
const videoTexture = new VideoTexture(video); 
const material = new ShaderMaterial({
        uniforms: {
          videoTexture: {
            value: `videoTexture,`
          },
          threshold: {
            value: 0.05,
          },
        },
        vertexShader: `
            varying vec2 vUv;
            void main() {
                vUv = uv;
                gl_Position = projectionMatrix * modelViewMatrix * vec4( position, 1.0 );
            }`,
        fragmentShader: `
        uniform sampler2D videoTexture;
        uniform float threshold;
        varying vec2 vUv;
        void main() {
            vec4 color = texture2D(videoTexture, vUv);
            float greenScreen = color.g - max(color.r, color.b);
            float alpha = 1.0 - smoothstep(threshold - 0.05, threshold + 0.05, greenScreen);
            if(color.r == 0.0 && color.b == 0.0 && color.g == 0.0) {
                alpha = 0.0;
            }
            gl_FragColor = vec4(color.rgb, alpha);
        }
        `,
        transparent: true,
      });
mesh.material = matetial
```
add this code the background of video markers is transparent

**Comment by mistic100** [comment]
- Do you have an example video ?
- Where does this code come from ? I am woried it is heavily opinionated, what about blue-screens for exemple ? Might be better to use shader with more options like https://sbcode.net/threejs/webcam/

**Comment by tjw-javen** [comment]
https://github.com/mistic100/Photo-Sphere-Viewer/assets/36066822/f37b97e0-1526-4b1c-963c-87f75c06cdf1

![image](https://github.com/mistic100/Photo-Sphere-Viewer/assets/36066822/5abfbfc3-7829-42e6-b225-76f55feba5f7)

**Comment by mistic100** [comment]
I'll try to implement a suitable shader.

**Comment by mistic100** [comment]
I released version `5.5.0-alpha.2` with this feature.

Documentation is here https://photo-sphere-viewer.js.org/plugins/markers.html#chromakey

And a demo https://photo-sphere-viewer.js.org/demos/markers/chroma-key.html
